StackExchange.Profiling/Timing.cs

```diff
@@ -81,21 +81,77 @@ public Timing(MiniProfiler profiler, Timing parent, string name, decimal? minSav
         /// Gets or sets All sub-steps that occur within this Timing step. Add new children through <see cref="AddChild"/>
         /// </summary>
         [DataMember(Order = 5)]
-        public List<Timing> Children { get; set; }
+        public List<Timing> Children
```

Comment:
hmm yeah, the ImmutableList perf isn't great because inside it's actually a tree.
But they have an ImmutableArray which is faster to read, but more expensive to write http://stackoverflow.com/a/28621409/830724

Anyway, I think perhaps the best thing to do is to have this return an IReadOnlyList or IReadOnlyCollection that way it is obvious that the caller shouldn't mutate it. If we really cared about performance we could return an ImmutableArray or IReadOnlyList  here and cache the result, so we don't have to create the copy on every access.